### PR TITLE
Added keywrapping, for easier scripting in AHK

### PR DESCRIPTION
**Pull request by legomannetje**
Since I wanted to use it with AHK, but didn't want to intercept all the key's, i modified the script slightly to be able to wrap the key in an other key.

The code is modified in such a way that i can be disabled by commenting 1 line of code.

### CODE REVIEW

**Comment by koriwi** [comment]
code looks fine.
can you clarify what it does with AHK?
As i understand it:
you specify a hotkey in AHK, after this hotkey is pressed, every key is intercepted until the specifc key is pressed again?

**Comment by koriwi** [comment]
can you please change your whitespaces to the original ones? you would change every line of code :D
and also can you disable Key wrapping by default please?

File: app/app.ino [hunk 1]
```diff
@@ -1,508 +1,532 @@
...
+
+  How it works:
+
+  Before pressing a key that you picked in the configurator, the freedeck will always press a special key.
+  I advice an F13 or higher key, since this will most likely not be on your normal keyboard to press by accident (And is supported by keyboard.h)
+
+  Command out the line #define KEYWRAPPING if you dont wanna use it!
+
+*/
+#define KEYWRAPPING
```
Comment (koriwi) [inline app/app.ino:51]: pls disable by default

**Comment by legomannetje** [comment]
Yes. First the freedeck pressed the F24 key (the wrap key) and then an other key. AHK can see that both keys are pressed, and therefore execute the marco action.

If it will be usefull i can also attach an AHK-script for simple modification with it

File: app/app.ino (hunk 1 above)
Comment (legomannetje) [reply app/app.ino]: Solved with commit 245b3eb

**Comment by legomannetje** [comment]
I've changed the line ending, but it does not seem do solve the issue, do you know how to fix that one?

**Comment by koriwi** [comment]
it's not about the line ending. its about tabs/spaces i believe. you don't use the arduino ide?

**Comment by koriwi** [comment]
could you upload the ahk script to this new repo? https://github.com/koriwi/freedeck-tools
it's nice to have the ahk script, but it does not belong in the arduino repo

**Comment by legomannetje** [comment]
I'm using the arduino IDE, is thats the problem?

And i'll create a push request for the ahk scripts in the other repo

**Comment by koriwi** [comment]
i'm using it, too. but on linux. so maybe you can check your line ending again and what your whitespaces are locally before commiting.

**Comment by koriwi** [comment]
looks good, now i just need to sort out the whitespace issues.

**Comment by legomannetje** [comment]
I've tried to put the F24 in the configurator, and this works just as well as the keywrapping done inside the code.
So this modification in the code is no needed for this functionality

**Comment by koriwi** [comment]
perfect!

**Pull request closed without merging**